### PR TITLE
[CBRD-25195] Added a SQL test case for A core dump occurs when multiple tables and serial functions are used simultaneously in one query

### PR DESCRIPTION
**Pull request by junsklee**
Refer to http://jira.cubrid.org/browse/CBRD-25195

### CODE REVIEW

File: sql/_13_issues/_24_1h/cases/cbrd_25195.sql [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+create table t20 (a int);
+
+--/* ----------------------------------------
+--*
+--* Test case 1 - min_tables_single_cast
+--* 
+--* Ensure core dump no longer occurs when using 10 tables and 1 serial function.
+--*     => Selects data from 9 tables -> select 1 serial function -> select 10th table;
+--* Expected Result: The query should return 1
+--* ---------------------------------------- */
```
Comment (kwonhoil) [inline sql/_13_issues/_24_1h/cases/cbrd_25195.sql:54]: select  'Test case 1 - min_tables_single_cast' ;

File: sql/_13_issues/_24_1h/cases/cbrd_25195.sql [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+
+--/* ----------------------------------------
+--*
+--* Test case 2 - min_tables_many_casts
+--* 
+--* Ensure core dump no longer occurs when using 10 tables and 10 serial functions sequentially.
+--*     => Selects data from 10 tables and 10 serial functions sequentially and returns the number of rows in the output 
+--* Expected Result: The query should return 10 results with values 1 
+--* ---------------------------------------- */
```
Comment (kwonhoil) [inline sql/_13_issues/_24_1h/cases/cbrd_25195.sql:91]: select 'Test case 2 - min_tables_many_casts';

File: sql/_13_issues/_24_1h/cases/cbrd_25195.sql [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+
+--/* ----------------------------------------
+--*
+--* Test case 3 - max_tables_single_cast
+--* 
+--* Ensure core dump no longer occurs when using 20 tables and 1 serial function.
+--*     => Selects data from 19 tables -> select 1 serial function -> select 20th table;
+--* Expected Result: The query should return 1
+--* ---------------------------------------- */
```
Comment (kwonhoil) [inline sql/_13_issues/_24_1h/cases/cbrd_25195.sql:164]: select  'Test case 3 - max_tables_single_cast' ;

File: sql/_13_issues/_24_1h/cases/cbrd_25195.sql [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+
+--/* ----------------------------------------
+--*
+--* Test case 4 - max_tables_many_casts
+--* 
+--* Ensure core dump no longer occurs when using 20 tables and 10 serial functions.
+--*     => Selects data from 20 tables and 10 serial functions sequentially and returns the number of rows in the output.
+--* Expected Result: The query should return 10 results with values 1 
+--* ---------------------------------------- */
```
Comment (kwonhoil) [inline sql/_13_issues/_24_1h/cases/cbrd_25195.sql:221]: select  'Test case 4 - max_tables_many_casts' ;

**Review by kwonhoil — APPROVED** [review]

**Review by ssihil — APPROVED** [review]

File: sql/_13_issues/_24_1h/cases/cbrd_25195.sql (hunk 2 above)
Comment (bagus-kim) [reply sql/_13_issues/_24_1h/cases/cbrd_25195.sql:91]: I only checked it in Test case 1.
1) it has nothing to do with next_values and current_values
- union all
- select cast(sr1.CURRENT_VALUE as int) 
2) It's not related to setting AUTO_INCREMENT
- create table t1 (a int AUTO_INCREMENT, id varchar);
3) It only happens in the 10th.
4) Also occurs if UNION

Please consider adding a case 3, 4

Comment (junsklee) [reply sql/_13_issues/_24_1h/cases/cbrd_25195.sql:91]: The suggested changes/additions have been reflected in the latest commit:
1. it has nothing to do with next_values and current_values ->  'Test case 6 - current_value_cast'
2. It's not related to setting AUTO_INCREMENT -> 'Test case 7 - auto_increment_table_cast'
3. It only happens in the 10th -> Test case 2,3,4
4. Also occurs if UNION -> 'Test case 5 - distinct_union_cast'

**Review by tw-kang — APPROVED** [review]

**Review by bagus-kim — APPROVED** [review]

**Pull request merged**